entity-service/src/test/kotlin/com/egm/stellio/entity/authorization/Neo4jAuthorizationRepositoryTest.kt

```diff
@@ -0,0 +1,214 @@
+                "urn:ngsi-ld:User:01",
+                listOf("urn:ngsi-ld:Apiary:01", "urn:ngsi-ld:Apiary:02"),
+                setOf(EGM_CAN_READ, EGM_CAN_WRITE)
+            )
+
+        assert(authorizedEntitiesId == listOf("urn:ngsi-ld:Apiary:01", "urn:ngsi-ld:Apiary:02"))
+
+        neo4jRepository.deleteEntity("urn:ngsi-ld:User:01")
+        neo4jRepository.deleteEntity("urn:ngsi-ld:Apiary:01")
+        neo4jRepository.deleteEntity("urn:ngsi-ld:Apiary:02")
+    }
+
+    @Test
+    fun `it should get all user's roles`() {
+        createEntity(
+            "urn:ngsi-ld:User:01",
+            listOf("User"),
+            mutableListOf(
+                Property(
+                    name = EGM_ROLES,
+                    value = listOf("admin", "creator")
+                )
+            )
+        )
+
+        val roles = neo4jAuthorizationRepository.getUserRoles("urn:ngsi-ld:User:01")
+
+        assert(roles == listOf("admin", "creator"))
+
+        neo4jRepository.deleteEntity("urn:ngsi-ld:User:01")
+    }
+
+    @Test
+    fun `it should get all user's roles from group`() {
+        val userEntity = createEntity("urn:ngsi-ld:User:01", listOf("User"), mutableListOf())
+
+        val groupEntity = createEntity(
+            "urn:ngsi-ld:Group:01",
+            listOf("Group"),
+            mutableListOf(
+                Property(
+                    name = EGM_ROLES,
+                    value = listOf("admin")
+                )
+            )
+        )
+
+        createRelationship(EntitySubjectNode(userEntity.id), EGM_IS_MEMBER_OF, groupEntity.id)
+
+        val roles = neo4jAuthorizationRepository.getUserRoles("urn:ngsi-ld:User:01")
+
+        assert(roles == listOf("admin"))
+
+        neo4jRepository.deleteEntity("urn:ngsi-ld:User:01")
+        neo4jRepository.deleteEntity("urn:ngsi-ld:Group:01")
+    }
+
+    @Test
+    fun `it should find no user roles`() {
+        createEntity("urn:ngsi-ld:User:01", listOf("User"), mutableListOf())
+
+        val roles = neo4jAuthorizationRepository.getUserRoles("urn:ngsi-ld:User:01")
+
+        assert(roles.isEmpty())
+
+        neo4jRepository.deleteEntity("urn:ngsi-ld:User:01")
+    }
+
+    @Test
+    fun `it should create admin links to entities`() {
+        createEntity("urn:ngsi-ld:User:01", listOf("User"), mutableListOf())
+        createEntity("urn:ngsi-ld:Apiary:01", listOf("Apiary"), mutableListOf())
+        createEntity("urn:ngsi-ld:Apiary:02", listOf("Apiary"), mutableListOf())
+
+        val targetIds = listOf("urn:ngsi-ld:Apiary:01", "urn:ngsi-ld:Apiary:02")
+
+        val createdRelations = neo4jAuthorizationRepository.createAdminLinks(
+            "urn:ngsi-ld:User:01",
+            targetIds.map {
+                Relationship(
+                    type = listOf(R_CAN_ADMIN),
+                    datasetId = URI.create("urn:ngsi-ld:Dataset:rCanAdmin:$it")
+                )
+            },
+            targetIds
+        )
+
+        assert(createdRelations.size == 2)
```

Comment:
Sadly this test should be improved to check if Relationships are correctly created (targetting entity and coming from User entity) but this is hard because search for relationships with datasetId is not yet implemented and required there.

Reply:
I still not get the reason behind the datasetId.
Even if the user can admin more than one entity, the target entity is not same, so why we use it ?

Reply:
Because we need to have different relationships with same name. 
When you want to delete a relationship, you do it from the source entity, meaning from the source entity you need a way which is common for both Property and Relationships to delete this attribute. As the target entity is only a specificity of Relationship, and the name is common, you need the datasetId.